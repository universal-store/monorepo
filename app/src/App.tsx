/** @format */

import React from 'react';

import { theme } from '&theme';
import { ThemeProvider } from 'styled-components/native';

import { ApolloClient, ApolloProvider, HttpLink, InMemoryCache } from '@apollo/client';

import { Root } from '&navigation';

// Create the client
const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: new HttpLink({
    uri: 'http://localhost:8080/v1/graphql',
  }),
});

const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <ApolloProvider client={client}>
        <Root />
      </ApolloProvider>
    </ThemeProvider>
  );
};

export default App;
