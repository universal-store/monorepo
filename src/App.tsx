/** @format */

import React from 'react';
import { StatusBar } from 'react-native';

import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';

// import { config } from "./configs";

// Create the client as outlined in the setup guide
const client = new ApolloClient({
  cache: new InMemoryCache(),
  // link: new HttpLink({
  // 	uri: config.API_URL,
  // }),
});
const App = () => {
  return (
    <ApolloProvider client={client}>
      <StatusBar barStyle="dark-content" />
    </ApolloProvider>
  );
};

export default App;
