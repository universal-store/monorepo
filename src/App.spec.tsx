import React from 'react';
import TestRenderer from 'react-test-renderer';

import App from './App';

const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms));

describe('App', () => {
  it('snapshot test', async () => {
    const tree = TestRenderer.create(
      <App skipLoadingScreen />,
    ).toJSON();
    await sleep(1);
    expect(tree).toMatchSnapshot();
  });
});
