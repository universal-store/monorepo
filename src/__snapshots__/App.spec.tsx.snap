// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
